site/_apps/elm-hn-pwa.txt

```diff
@@ -0,0 +1,25 @@
+---
+weight: 24 
+title: 'Elm HN PWA'
+github-title: 'skrypte/elm-hn-pwa'
+libraries: "Elm" 
+scaffolding: 'N/A'
+module-bundling: 'N/A'
+service-worker: 'partially workbox.js + application shell + data caching'
+perfomance-patterns: 'preconnect API, inline CSS, minified JS, lazy loading for JSON data'
+server-side-rendering: 'None'
+api: 'hacker-news.firebaseio.com/v0'
+hosting: 'o2switch'
+authors: 
+    - name: 'skingrapher'
+lighthouse: '91/100'
+interactive-em: '4.8s'
+interactive-faster-3g: '3.7s'
+lighthouse-link: https://www.webpagetest.org/lighthouse.php?test=171104_BM_ce030e66965f4ea601c73316b8448f03&run=1
+wpt-em-link: https://www.webpagetest.org/result/171104_BM_ce030e66965f4ea601c73316b8448f03/ 
+wpt-faster-3g-link: https://www.webpagetest.org/result/171104_NK_697bd1b5a53d8b3ef8e0afd8a6579a3e/ 
```

Comment:
Our general expectations are that data should be fetched and the application is interactive (both the main thread JavaScript and network have settled) within ~5/6 seconds. Looking at WPT, even 20+ seconds in, we're still not getting interactive.

![image](https://user-images.githubusercontent.com/110953/33106498-298b2a40-cee7-11e7-9da0-1da6b24c2584.png)

Needing to do a separate query for every JSON file appears to be the issue here. There are a lot of them:

![image](https://user-images.githubusercontent.com/110953/33106546-6888fc18-cee7-11e7-9063-84271cf12397.png)

Reply:
the issue is here. need to change the code